dnf5-plugins/CMakeLists.txt

```diff
@@ -6,3 +6,4 @@ include_directories("${PROJECT_SOURCE_DIR}/dnf5/include/")
 
 add_subdirectory("changelog_plugin")
 add_subdirectory("builddep_plugin")
+add_subdirectory("copr_plugin")
```

Comment:
Does it make sense to have it here (dnf5-plugins)?  Or do you think we could make it a default dnf5 command (non-plugin)?  We don't expect any additional third-party dependencies that would enlarge the default minimal installations.

Reply:
@praiskup Good question. I don't want to answer yes or no now. But something to think about.
For example, the `install` command is a general command. Usable by everyone regardless of Linux distribution.
What about the `copr` command? Is this a general command? Or is it specific? For example, for users of a specific Linux distribution (eg Fedora). What will be the benefit of having it installed everywhere as a default command?

Reply:
It is mostly usable by anyone I think (but my perspective might be different). Copr provides a wide variety of chroots potentially consumable on any RPM distribution (Mageia/OpenSUSE/Fedora/...).  In some situations it is viable to enable "fedora-rawhide-x86_64" on "any-x86_64" host, because the packages are noarch and installable anywhere.

Benfit is clear;  closer integration with Copr (e.g. in `Containerfile` there's no need to `dnf5 install -y dnf5-plugins` first, you go directly to `dnf5 copr enable alice/project`).

Reply:
This is really tricky.  The most important part is related to containers and minimal images. It provides functionality for managing repositories not delivered by RPMs. I can use copr repositories without copr plugin, therefore I believe that it is not an essential functionality, but enhancement. Somehow I have a feeling that it would be good to keep it as a plugin.

Reply:
From my POW it is better to have all non-dnf related features in a plugin. It makes things more granular and with a direct way to distinguish which is a crucial part of DNF and what is not. Also if ever COPR would be discontinued this will not affect DNF in any way. DNF is a package manager and a library first and should be build system agnostic.

Reply:
Accepted, let's have it as plugin (and perhaps revamp the idea in the future?).

Just a small correction, Copr is not just a build system, it is a place where projects grow.
Especially from this plugin perspective where we solve "how to consume project content"
(definitely not "how to build").  The functionality is close to the configmanager logic.

Copr is not just Fedora Copr;  there are other instances out there already - and more
deployments are likely to appear because deployment is easier nowadays.